### DISCUSSION TITLE
How to get confusion matrix at each epoch

### DISCUSSION THREAD

**Discussion by AlainPilon** [ultralytics/ultralytics#5771] [Q&A]
As the title says, when training a classifier, I would like to see a confusion matrix at the end of each epoch. We "need" this to see the behaviour of our training against our very specific data set as we increase the number of epoch. 

Is there an hidden option to do this?

**Comment by glenn-jocher** [comment]
@AlainPilon hello! Currently, YOLOv8 does not provide an out-of-the-box feature to display a confusion matrix at the end of each epoch during training. However, the validation results at the end of each epoch include precision, recall, and mAP metrics that can help you understand the performance of your model on your specific dataset.

If you require a confusion matrix for each epoch, you would need to implement custom code that utilizes the validation predictions and ground truth annotations to construct the matrix after each training epoch. This could be done by modifying the validation routine in the source code.

For further guidance on how to interpret the existing validation metrics or how to modify the code for additional features, you can refer to our documentation at https://docs.ultralytics.com. Keep in mind that any modifications to the source code would be your responsibility to maintain and ensure compatibility with future updates.

Thank you for your contribution to the YOLO community, and best of luck with your training! 🚀
